packages/wxt/src/storage.ts

```diff
@@ -461,7 +461,16 @@ export interface WxtStorage {
    * @example
    * await storage.getItem<number>("local:installDate");
    */
-  getItem<T>(key: StorageItemKey, opts?: GetItemOptions<T>): Promise<T | null>;
+  getItem<TValue>(
+    key: StorageItemKey,
+    opts: GetItemOptions<TValue> & { fallback: TValue },
+  ): Promise<TValue>;
+
+  getItem<TValue>(
+    key: StorageItemKey,
+    opts?: GetItemOptions<TValue>,
+  ): Promise<TValue | null>;
```

Comment:
I'm pretty sure that to solve #1005 it needs to return `Promise<TValue | null>`

Reply:
@avi12 The issue the pull request has solved is the proper type hinting. `null` is more suitable option as this tell the developer explicitly that the is **"no value"** and we should leave `undefined` to JavaScript itself as that indicates something unexpected has happened.

As a developer, we should always go for `null` and JS core will default to `undefined` and this is really important when finding out where the issue arose as one is by developer and another is my accident. Hope you get that now.

Thanks.

Reply:
I guess my alternative can be
```ts
const [value1 = "foo", value2 = "bar"] = await Promise.all([
  storage.getItem<string>("local:key1"),
  storage.getItem<string>("local:key2")
]);
```

Reply:
Your alternative would only work if the get item function returns undefined, otherwise, the code you wrote would not work as the get function would return for the item you are trying to get in the value for your variables would also be null.

Reply:
You're right, my alternative can only be
```ts
const [value1 = "foo", value2 = "bar"] = await Promise.all([
  storage.getItem<string>("local:key1"),
  storage.getItem<string>("local:key2")
]) as [string, string];
```

Reply:
This won't work as the value you provided after _=_ sign will only be assigned if the value returned by promise is undefined. The promise returns a null so the value1 would be null not the string you specified as fallback with equals sign.

Reply:
My point is that `getItem` not returning `undefined` is very unfortunate for my use case and I'm sure I'm not the only one who holds this opinion

Reply:
That is what I am saying return `undefined` should be only kept for compiler as a developer we should return `null`

Reply:
I don't think the Chromium team agrees with you, considering that
```ts
const { key1 } = await chrome.storage.local.get("key1");
```
will either have `undefined` or a non-`undefined` value, which lets me do
```ts
const { key1 = "foo" } = await chrome.storage.local.get("key1");
```
which is very maintainable and readable for me

Reply:
It's up to the developers liking, I am no one to specify what is right or wrong. The repo owner can guide us further. Thanks.